Problem with typescript

```
node_modules/vue3-apexcharts/typings/vue-apexcharts.d.ts:2:30 - error TS2307: Cannot find module 'vue/types/plugin' or its corresponding type declarations.

2 import { PluginObject } from 'vue/types/plugin';
                               ~~~~~~~~~~~~~~~~~~

node_modules/vue3-apexcharts/typings/vue-apexcharts.d.ts:5:42 - error TS2709: Cannot use namespace 'Vue' as a type.

5 interface VueApexChartsComponent extends Vue {
                                           ~~~

node_modules/vue3-apexcharts/typings/vue-apexcharts.d.ts:36:59 - error TS2709: Cannot use namespace 'Vue' as a type.

36 declare const VueApexCharts: Component & ComponentOptions<Vue> & PluginObject<any>;
            
```

Thanks for the great project! Unfortunately I've been unable to use it due to this error. Any recommendations?

--- a/README.md
+++ b/README.md
@@ -222,7 +222,6 @@ You don't actually need to call updateSeries() or updateOptions() manually. Chan
 | <a href="https://apexcharts.com/docs/methods/#updateOptions">updateOptions</a>           | Allows you to update the configuration object                                                      |
 | <a href="https://apexcharts.com/docs/methods/#toggleSeries">toggleSeries</a>             | Allows you to toggle the visibility of series programatically. Useful when you have custom legend. |
 | <a href="https://apexcharts.com/docs/methods/#appendData">appendData</a>                 | Allows you to append new data to the series array.                                                 |
-| <a href="https://apexcharts.com/docs/methods/#addtext">addText</a>                       | The addText() method can be used to draw text after chart is rendered.                             |
 | <a href="https://apexcharts.com/docs/methods/#addxaxisannotation">addXaxisAnnotation</a> | Draw x-axis annotations after chart is rendered.                                                   |
 | <a href="https://apexcharts.com/docs/methods/#addyaxisannotation">addYaxisAnnotation</a> | Draw y-axis annotations after chart is rendered.                                                   |
 | <a href="https://apexcharts.com/docs/methods/#addpointannotation">addPointAnnotation</a> | Draw point (xy) annotations after chart is rendered.                                               |

--- a/README.md
+++ b/README.md
@@ -47,6 +47,21 @@ export default {
 };
 ```
 
+To provide a `$apexcharts` reference inside Vue instance
+
+```ts
+import ApexCharts from 'apexcharts';
+
+app.config.globalProperties.$apexcharts = ApexCharts;
+
+// Add this when into a TypeScript codebase
+declare module '@vue/runtime-core' {
+  interface ComponentCustomProperties {
+    $apexcharts: typeof ApexCharts;
+  }
+}
+```
+
 To create a basic bar chart with minimal configuration, write as follows:
 
 ```vue

--- a/src/vue3-apexcharts.js
+++ b/src/vue3-apexcharts.js
@@ -3,6 +3,7 @@ import {
   h,
   defineComponent,
   ref,
+  getCurrentInstance,
   onMounted,
   onBeforeUnmount,
   watch,
@@ -56,10 +57,10 @@ const vueApexcharts = defineComponent({
   },
 
   // events emitted by this component
-  emits: [...events],
+  emits: events,
 
   setup(props, { emit }) {
-    const el = ref(null);
+    const __el = ref(null);
     const chart = ref(null);
 
     const isObject = item => {
@@ -133,7 +134,7 @@ const vueApexcharts = defineComponent({
       });
 
       const config = extend(props.options, newOptions);
-      chart.value = new ApexCharts(el.value, config);
+      chart.value = new ApexCharts(__el.value, config);
       return chart.value.render();
     };
 
@@ -182,6 +183,10 @@ const vueApexcharts = defineComponent({
       return chart.value.appendData(newData);
     };
 
+    const zoomX = (start, end) => {
+      return chart.value.zoomX(start, end);
+    };
+
     const addText = options => {
       chart.value.addText(options);
     };
@@ -219,6 +224,7 @@ const vueApexcharts = defineComponent({
     });
 
     onMounted(() => {
+      __el.value = getCurrentInstance().proxy.$el;
       init();
     });
 
@@ -262,13 +268,33 @@ const vueApexcharts = defineComponent({
       refresh();
     });
 
-    return { el, chart };
+    return {
+      chart,
+      init,
+      refresh,
+      destroy,
+      updateOptions,
+      updateSeries,
+      toggleSeries,
+      showSeries,
+      hideSeries,
+      resetSeries,
+      zoomX,
+      toggleDataPointSelection,
+      appendData,
+      appendSeries,
+      addXaxisAnnotation,
+      addYaxisAnnotation,
+      addPointAnnotation,
+      removeAnnotation,
+      clearAnnotations,
+      dataURI
+    };
   },
 
   render() {
     return h("div", {
       class: "vue-apexcharts",
-      ref: "el"
     });
   }
 });

--- a/src/vue3-apexcharts.js
+++ b/src/vue3-apexcharts.js
@@ -187,10 +187,6 @@ const vueApexcharts = defineComponent({
       return chart.value.zoomX(start, end);
     };
 
-    const addText = options => {
-      chart.value.addText(options);
-    };
-
     const dataURI = () => {
       return chart.value.dataURI();
     };
@@ -282,7 +278,6 @@ const vueApexcharts = defineComponent({
       zoomX,
       toggleDataPointSelection,
       appendData,
-      addText,
       appendSeries,
       addXaxisAnnotation,
       addYaxisAnnotation,

--- a/src/vue3-apexcharts.js
+++ b/src/vue3-apexcharts.js
@@ -282,6 +282,7 @@ const vueApexcharts = defineComponent({
       zoomX,
       toggleDataPointSelection,
       appendData,
+      addText,
       appendSeries,
       addXaxisAnnotation,
       addYaxisAnnotation,

--- a/typings/vue-apexcharts.d.ts
+++ b/typings/vue-apexcharts.d.ts
@@ -1,8 +1,7 @@
-import Vue, { Component, ComponentOptions } from 'vue';
-import { PluginObject } from 'vue/types/plugin';
 import ApexCharts, { ApexOptions } from 'apexcharts';
+import { Component, ComponentOptions, ComponentPublicInstance, Plugin } from 'vue';
 
-interface VueApexChartsComponent extends Vue {
+export interface VueApexChartsComponent extends ComponentPublicInstance {
   // data
   readonly chart?: ApexCharts;
   // props
@@ -24,6 +23,7 @@ interface VueApexChartsComponent extends Vue {
   zoomX(min: number, max: number): void;
   toggleDataPointSelection(seriesIndex: number, dataPointIndex?: number): any;
   appendData(newData: any): Promise<void>;
+  addText(text: string): void;
   appendSeries(newSeries: any, animate?: boolean): Promise<void>;
   addXaxisAnnotation(options: any, pushToMemory?: boolean, context?: any): void;
   addYaxisAnnotation(options: any, pushToMemory?: boolean, context?: any): void;
@@ -33,12 +33,6 @@ interface VueApexChartsComponent extends Vue {
   dataURI(): Promise<string>;
 }
 
-declare const VueApexCharts: Component & ComponentOptions<Vue> & PluginObject<any>;
+declare const VueApexCharts: Component & ComponentOptions<VueApexChartsComponent> & Plugin;
 
 export default VueApexCharts;
-
-declare module 'vue/types/vue' {
-  interface Vue {
-    $apexcharts: typeof ApexCharts;
-  }
-}

--- a/typings/vue-apexcharts.d.ts
+++ b/typings/vue-apexcharts.d.ts
@@ -23,7 +23,6 @@ export interface VueApexChartsComponent extends ComponentPublicInstance {
   zoomX(min: number, max: number): void;
   toggleDataPointSelection(seriesIndex: number, dataPointIndex?: number): any;
   appendData(newData: any): Promise<void>;
-  addText(text: string): void;
   appendSeries(newSeries: any, animate?: boolean): Promise<void>;
   addXaxisAnnotation(options: any, pushToMemory?: boolean, context?: any): void;
   addYaxisAnnotation(options: any, pushToMemory?: boolean, context?: any): void;

